src/PFire.Core/Protocol/XMessageField.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System;
+using System.Text;
+
+namespace PFire.Protocol
+{
+    [AttributeUsage(AttributeTargets.Field | AttributeTargets.Method | AttributeTargets.Property, AllowMultiple = true)]
+    internal sealed class XMessageField : Attribute
+    {
+        public string Name { get; }
+        public byte[] NameAsBytes { get; }
```

Comment:
A small concern with this is someone can accidentally diverage the value between `Name` and `NameAsBytes`. Although less performant, I'd prefer `NameAsBytes => Encoding.UTF8.GetBytes(name)`

Reply:
Yep, good call. I have changed NameAsBytes to your suggestion.